src/singletons.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import type { PathType, PropertiesType } from "./types.js";
+
+export let Path: PathType = (null: any);
+export let Properties: PropertiesType = (null: any);
+
+export function setPath(singleton: PathType) {
+  Path = singleton;
+}
+
+export function setProperties(singleton: PropertiesType) {
```

Comment:
Is there really a need for these helper functions `setPath`, `setProperties`? They mutate an exported variable, so the caller could just do that directly.

Reply:
That was my first thought too, but Flow does not like it:

18:   Singletons.Path = new PathImplementation();
       ^^^^^^^^^^^^^^^ assignment of property `Path`. Mutation not allowed on
 18:   Singletons.Path = new PathImplementation();
       ^^^^^^^^^^ exports of "./singletons.js"